test/test_tools.py

```diff
@@ -96,3 +107,256 @@ def _minimum_is_zero(array: np.ndarray) -> bool:
+def test_separate_real_imag_of_vector_first_half(vector):
+    assert_equal(separate_real_imag_of_vector(vector)[0], vector[: len(vector) // 2])
+
+
+@given(hnp.arrays(dtype=hnp.scalar_dtypes(), shape=hypothesis_even_dimension()))
+def test_separate_real_imag_of_vector_second_half(vector):
+    assert_equal(separate_real_imag_of_vector(vector)[1], vector[len(vector) // 2 :])
+
+
+@given(
+    hnp.arrays(
+        dtype=hnp.scalar_dtypes(),
+        shape=hst.builds(
+            lambda shape_tuple: (shape_tuple[0], shape_tuple[1] * 2 - 1),
+            hnp.array_shapes(min_dims=2, max_dims=2, max_side=10),
+        ),
+    )
+)
+def test_separate_real_imag_of_mc_samples_wrong_len(array):
+    with pytest.raises(
+        ValueError,
+        match="separate_real_imag_of_mc_samples: vectors of real and imaginary "
+        "parts are expected to contain exactly as many real as "
+        r"imaginary parts but the first one is of odd length=.*",
+    ):
+        separate_real_imag_of_mc_samples(array)
+
+
+@given(
+    hnp.arrays(
+        dtype=hnp.scalar_dtypes(),
+        shape=hst.builds(
+            lambda shape_tuple: (shape_tuple[0], shape_tuple[1] * 2),
+            hnp.array_shapes(min_dims=2, max_dims=2, max_side=10),
+        ),
+    )
+)
+def test_separate_real_imag_of_mc_samples_dimensions(array):
+    list_of_separated_real_imag = separate_real_imag_of_mc_samples(array)
+    assert_equal(len(list_of_separated_real_imag), 2)
+    assert_equal(
+        _set_of_shapes_of_ndarray_list(list_of_separated_real_imag),
+        {(len(array), len(array[0]) / 2)},
+    )
+
+
+def _set_of_shapes_of_ndarray_list(
+    list_of_anything: List[np.ndarray],
+) -> Set[Tuple[int, int]]:
+    return set(list_entry.shape for list_entry in list_of_anything)
+
+
+@given(
+    hnp.arrays(
+        dtype=hnp.scalar_dtypes(),
+        shape=hst.builds(
+            lambda shape_tuple: (shape_tuple[0], shape_tuple[1] * 2),
+            hnp.array_shapes(min_dims=2, max_dims=2, max_side=10),
+        ),
+    )
+)
+def test_separate_real_imag_of_mc_samples_first_half(array):
+    assert_equal(
+        separate_real_imag_of_mc_samples(array)[0], array[:, : len(array[0]) // 2]
+    )
+
+
+@given(
+    hnp.arrays(
+        dtype=hnp.scalar_dtypes(),
+        shape=hst.builds(
+            lambda shape_tuple: (shape_tuple[0], shape_tuple[1] * 2),
+            hnp.array_shapes(min_dims=2, max_dims=2, max_side=10),
+        ),
+    )
+)
+def test_separate_real_imag_of_mc_samples_second_half(array):
+    assert_equal(
+        separate_real_imag_of_mc_samples(array)[1], array[:, len(array[0]) // 2 :]
+    )
+
+
+@given(
+    hnp.arrays(
+        dtype=hst.one_of(
+            hnp.unsigned_integer_dtypes(), hnp.integer_dtypes(), hnp.floating_dtypes()
+        ),
+        shape=hst.builds(
+            lambda shape_tuple: (shape_tuple[0], shape_tuple[1] * 2),
+            hnp.array_shapes(min_dims=2, max_dims=2, max_side=10),
+        ),
```

Comment:
I think it would be useful to extract this into its own strategy.

Reply:
Resolved by 173481a4 and 48ab4a73